resnet.py

```diff
@@ -154,7 +169,7 @@ def build_resnet_152(input_shape, num_outputs):
 
 
 def main():
-    model = ResNetBuilder.build_resnet_18((3, 224, 224), 1000)
+    model = ResNetBuilder.build_resnet_18(INPUT_SHAPE, 1000)
```

Comment:
Thanks for the PR.
build method always expects input of shape (nb_channels, nb_rows, nb_cols) per docs. You could handle backend differences by rolling axis inside this method instead.